WordPress/src/main/java/org/wordpress/android/ui/accounts/login/LoginNoSitesFragment.kt

```diff
@@ -0,0 +1,152 @@
+package org.wordpress.android.ui.accounts.login
+
+import android.content.Context
+import android.os.Bundle
+import android.view.View
+import androidx.activity.OnBackPressedCallback
+import androidx.fragment.app.Fragment
+import androidx.lifecycle.ViewModelProvider
+import org.wordpress.android.R
+import org.wordpress.android.WordPress
+import org.wordpress.android.databinding.FragmentLoginNoSitesEmptyViewBinding
+import org.wordpress.android.login.LoginListener
+import org.wordpress.android.login.util.AvatarHelper
+import org.wordpress.android.login.util.AvatarHelper.AvatarRequestListener
+import org.wordpress.android.ui.ActivityLauncher
+import org.wordpress.android.ui.accounts.LoginNavigationEvents.ShowInstructions
+import org.wordpress.android.ui.accounts.LoginNavigationEvents.ShowSignInForResultJetpackOnly
+import org.wordpress.android.ui.accounts.login.LoginNoSitesViewModel.State.NoUser
+import org.wordpress.android.ui.accounts.login.LoginNoSitesViewModel.State.ShowUser
+import org.wordpress.android.ui.main.utils.MeGravatarLoader
+import org.wordpress.android.ui.utils.UiHelpers
+import javax.inject.Inject
+
+@Suppress("TooManyFunctions")
+class LoginNoSitesFragment : Fragment(R.layout.fragment_login_no_sites_empty_view) {
+    companion object {
+        const val TAG = "LoginNoSitesFragment"
+
+        fun newInstance(): LoginNoSitesFragment {
+            return LoginNoSitesFragment()
+        }
+    }
+
+    @Inject lateinit var viewModelFactory: ViewModelProvider.Factory
+    @Inject lateinit var meGravatarLoader: MeGravatarLoader
+    @Inject lateinit var uiHelpers: UiHelpers
+    private var loginListener: LoginListener? = null
+    private lateinit var viewModel: LoginNoSitesViewModel
+
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        super.onViewCreated(view, savedInstanceState)
+        initDagger()
+        initBackPressHandler()
+        with(FragmentLoginNoSitesEmptyViewBinding.bind(view)) {
```

Comment:
Praise (❤️): I really like what you did here, using extension functions with view bindings like that.